src/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -400,59 +429,62 @@ end
                                                v_neighbor_system, neighborhood_search)
-    (; pressure, cache, viscosity) = boundary_model
+    (; pressure, cache, viscosity, density_calculator) = boundary_model
+    (; pressure_offset) = density_calculator
 
     # Loop over all pairs of particles and neighbors within the kernel cutoff.
     for_particle_neighbor(system, neighbor_system,
                           system_coords, neighbor_coords,
                           neighborhood_search;
                           particles=eachparticle(system)) do particle, neighbor,
                                                              pos_diff, distance
-        adami_pressure_inner!(boundary_model, system, neighbor_system,
+        adami_pressure_inner!(boundary_model, system, neighbor_system, v,
                               v_neighbor_system, particle, neighbor, pos_diff,
-                              distance, viscosity, cache, pressure)
+                              distance, viscosity, cache, pressure, pressure_offset)
     end
 end
 
 @inline function adami_pressure_extrapolation!(boundary_model, system, neighbor_system,
-                                               system_coords, neighbor_coords,
+                                               system_coords, neighbor_coords, v,
                                                v_neighbor_system, neighborhood_search)
     return boundary_model
 end
 
 @inline function adami_pressure_inner!(boundary_model, system,
-                                       neighbor_system::FluidSystem,
+                                       neighbor_system::FluidSystem, v,
                                        v_neighbor_system, particle, neighbor, pos_diff,
-                                       distance, viscosity, cache, pressure)
+                                       distance, viscosity, cache, pressure,
+                                       pressure_offset)
     density_neighbor = particle_density(v_neighbor_system, neighbor_system, neighbor)
 
-    resulting_acc = neighbor_system.acceleration -
-                    current_acceleration(system, particle)
-
     kernel_weight = smoothing_kernel(boundary_model, distance)
 
-    pressure[particle] += (particle_pressure(v_neighbor_system, neighbor_system,
+    pressure[particle] += (pressure_offset +
+                           particle_pressure(v_neighbor_system, neighbor_system,
                                              neighbor) +
-                           dot(resulting_acc, density_neighbor * pos_diff)) *
-                          kernel_weight
+                           dynamic_pressure(density_neighbor, v, v_neighbor_system,
+                                            particle, neighbor, system)
+                           +
+                           dot(neighbor_system.acceleration,
+                               density_neighbor * pos_diff)) * kernel_weight
```

Comment:
Why? Is there a source for this? Does Adami do that somewhere?

Reply:
It might make sense to add a complete new `density_calculator` for this, no? I mean, Adami derived this equation with a force balance at the wall interface and not with the bernoulli equation.